arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/NonEmptySet.kt

```diff
@@ -43,20 +44,26 @@ public value class NonEmptySet<out A> private constructor(
   public override fun <B> flatMap(transform: (A) -> NonEmptyCollection<B>): NonEmptyList<B> =
     NonEmptyList(elements.flatMap(transform))
 
-  public override fun <B> mapIndexed(transform: (index:Int, A) -> B): NonEmptyList<B> =
+  public override fun <B> mapIndexed(transform: (index: Int, A) -> B): NonEmptyList<B> =
     NonEmptyList(elements.mapIndexed(transform))
 
   override fun <B> zip(other: NonEmptyCollection<B>): NonEmptyList<Pair<A, B>> =
     NonEmptyList(elements.zip(other))
+
+  internal companion object {
+    @JvmSynthetic
+    internal inline fun <A> unsafeOf(iterator: Iterator<A>): NonEmptySet<A> =
+      NonEmptySet(Iterable { iterator }.toSet())
+  }
 }
 
 public fun <A> nonEmptySetOf(first: A, vararg rest: A): NonEmptySet<A> =
-  NonEmptySet(first, rest.toSet())
+  NonEmptySet(first, rest.asList())
 
 public fun <A> Iterable<A>.toNonEmptySetOrNull(): NonEmptySet<A>? {
   val iter = iterator()
   if (!iter.hasNext()) return null
-  return NonEmptySet(iter.next(), Iterable { iter }.toSet())
+  return NonEmptySet.unsafeOf(iter)
```

Comment:
I don't think there's a need to have a `unsafeOf` if it's only used once, inline, and internal.

```suggestion
  return NonEmptySet(iter)
```

Reply:
Thanks, I have just changed it to `return NonEmptySet(iter.next(), Iterable { iter })` 🙏